nix/buildLeanPackage.nix

```diff
@@ -8,6 +8,10 @@ lib.makeOverridable (
   deps ? [ lean.Lean lean.Leanpkg ],
   # Static library dependencies. Each derivation `static` should contain a static library in the directory `${static}`.
   staticLibDeps ? [],
+  # Shared library dependencies included at interpretation with --load-dynlib and linked to. Each derivation `shared` should contain a 
+  # shared library at the path `${shared}/${shared.libName or shared.name}` and a name to link to like `-l${shared.linkName or shared.name}`.
+  # These libs are also linked to in packages that depend on this one.
+  nativeSharedLibs ? [],
```

Comment:
The "native" feels redundant

Reply:
Yes, I was unsure about this, but it is to distinguish it from shared libs that are written in lean. Do you want me to remove the native prefix?

Reply:
@Anderssorby there is no difference though? The shared library loaded through `--load-dynlib` *should* be a shared library written in Lean (though likely including functions with `@[extern]` definitions).

Reply:
I see your point, but one can link against shared lean libraries with the `deps` option which is what currently is the `leanSharedLibs` which includes recursive dependency loading.
The setup is a bit confusing, though, because you can either load things into the IR interpreter engine or link to it in a resulting library or executable.

Reply:
@Anderssorby To clarify, my point was that you want to link to the same native shared library both in `--load-dynlib` (for the interpreter) and in the actual build (of the shared Lean library or binary for the package and its dependents). However,, you also want to `--load-dynlib` the shared Lean libraries utilizing those native libraries.

 For instance, in your Blake3 package, there would be to options: 1) statically link the native blake3 to the shared Lean library for the Blake3 module and `--load-dynlib` just that (i.e., no native shared libraries), or 2) dynamically link the native blake3 library to the shared Lean library for Blake3 and `--load-dynlib` both (i.e., just the native library would be linked to the module, but both would be passed to `--load-dynlib`). 

Dependent package would then also need to respect this behavior (`--load-dynlib` the same libraries and link them to their shared libraries or binaries).

Reply:
@tydeu Correct me if I'm wrong but I think what you are mentioning is already satisfied (option 2).

Reply:
@Anderssorby oh, sorry, I was going off the discussion below and hadn't notice that you had updated the building to include the shared libraries.